R/get_gene_expression.R

```diff
@@ -1,294 +1,176 @@
-        wide_expression_data = fread(cmd = grep_cmd)
-      }
-      
-      # keep only the specified genes
-      if(! engine == "grep"){
-        wide_expression_data = wide_expression_data %>% 
-          dplyr::filter(.[[gene_id_type]] %in% gene_ids)
-      }
-      
-      wide_expression_data = wide_expression_data %>%
-        dplyr::select(-all_of(not_the_id_type)) %>%
-        group_by(mrna_sample_id, !!as.symbol(gene_id_type)) %>% #deal with non 1:1 mapping of Hugo to Ensembl
-        slice_head() %>%
-        as.data.frame() %>%
-        pivot_wider(names_from = !!as.symbol(gene_id_type), values_from = expression)
-    }
-  }
-  
-  # join the expression data to the sample ids of the metadata
-  if(missing(expression_data)){
-    join_with = ifelse(is.null(join_with), "NULL", join_with)
-    
-    if(join_with == "NULL" | join_with == "genome" | join_with == "capture"){
-      these_samples_metadata = dplyr::select(these_samples_metadata, sample_id, 
-                                             patient_id, biopsy_id, seq_type)
-      expression_wider = left_join(these_samples_metadata, wide_expression_data,
-                                   by = c("patient_id", "biopsy_id"))
-      
-      # add column `multi_exp` to inform whether there are more than one 
-      # `mrna_sample_id` associated to a `sample_id`
-      expression_wider = filter(expression_wider, !is.na(mrna_sample_id)) %>% 
-        { split(.$mrna_sample_id, .$sample_id) } %>% 
-        .[lengths(.) > 1] %>% 
-        lapply(unique) %>% 
-        .[lengths(.) > 1] %>% 
-        names %>% 
-        { mutate(expression_wider, multi_exp = ifelse(sample_id %in% ., 1, 0)) }
-      
-      ### filter out duplicated expressions based on prioritize_rows_by
-      if( !missing(prioritize_rows_by) & any(expression_wider$multi_exp == 1) ){
+          gettextf("grep -h -w -F -e %s %s", ., sample_files)
+        if(verbose){
+          print(grep_cmd)
+        }
         
-        # take only duplicated gene expression rows and split them by sample_id
-        multi_exp_split = dplyr::filter(expression_wider, multi_exp == 1) %>% 
-          split(.$sample_id)
-        
-        # use the first vector of the `prioritize_rows_by` list for the filtering. 
-        # if any duplication remains, use the next vector, and so on.
-        multi_exp_split = lapply(multi_exp_split, function(multi_exp_split_i){
-          for(based_column in names(prioritize_rows_by)){
-            for( prioritize_this_value in prioritize_rows_by[[based_column]] ){
-              k = multi_exp_split_i[,based_column] == prioritize_this_value
-              if( any(k) ){
-                multi_exp_split_i = multi_exp_split_i[k,]
-                break
-              } # else:
-              # if there is no row with this value to prioritize, keep the rows
-              # and try the lower-priority value of the next loop.
-            }
+        all_rows = fread(cmd = grep_cmd,col.names = c("ensembl_gene_id",
```

Comment:
Can you make this `data.table::fread` so that this works if the user doesn't have `data.table` loaded in their environment?